spec/models/crop_spec.rb

```diff
@@ -419,164 +419,106 @@
 
     context "scientific names" do
       it "adds a scientific name to a crop that has none" do
-        tomato = FactoryGirl.create(:tomato)
-        expect(tomato.scientific_names.size).to eq 0
-        tomato.add_scientific_names_from_csv("Foo bar")
+        row = ["parent", "http://en.wikipedia.org/wiki/Parent", "", "Foo bar"]
+        tomato = CsvImporter.new.import_crop(row)
         expect(tomato.scientific_names.size).to eq 1
         expect(tomato.default_scientific_name).to eq "Foo bar"
       end
 
       it "picks up scientific name from parent crop if available" do
-        parent = FactoryGirl.create(:crop, name: 'parent crop')
-        parent.add_scientific_names_from_csv("Parentis cropis")
-        parent.save
-        parent.reload
+        parent = CsvImporter.new.import_crop(
+          ["parent", "http://en.wikipedia.org/wiki/Parent", "", "Parentis cropis"])
+
+        tomato = CsvImporter.new.import_crop(
+          ["Tomato", "http://en.wikipedia.org/wiki/Parent", "parent"])
 
-        tomato = FactoryGirl.create(:tomato, parent: parent)
         expect(tomato.parent).to eq parent
         expect(tomato.parent.default_scientific_name).to eq "Parentis cropis"
-
-        tomato.add_scientific_names_from_csv('')
         expect(tomato.default_scientific_name).to eq "Parentis cropis"
       end
 
-      it "doesn't add a duplicate scientific name" do
-        tomato = FactoryGirl.create(:tomato)
```

Comment:
It looks like we're no longer testing that adding a (scientific|alternate) name is idempotent; this makes me nervous. How about a couple of tests along the following lines?

```ruby
it "doesn't add a duplicate scientific name" do
  row = ["parent", "http://en.wikipedia.org/wiki/Parent", "", "Foo bar, Foo bar"]
  tomato = CsvImporter.new.import_crop(row)
  expect(tomato.scientific_names.size).to eq 1
end

it "doesn't add a duplicate scientific name from parent" do
  parent = CsvImporter.new.import_crop(
    ["parent", "http://en.wikipedia.org/wiki/Parent", "", "Parentis cropis"])
  tomato = CsvImporter.new.import_crop(
    ["Tomato", "http://en.wikipedia.org/wiki/Parent", "parent". "Parentis cropis"])
  expect(tomato.scientific_names.size).to eq 1
end
```